### PR TITLE
ensure vignettes are opened on request (closes #3443)

### LINKED ISSUE #3443: attempts to open vignettes from Help pane prompt for download
### System details

    RStudio Edition : Desktop
    RStudio Version : v1.2.952
    OS Version      : macOS 10.13.6
    R Version       : 3.5.1

### Steps to reproduce the problem

Open the Rcpp help index with `help(package = "Rcpp")`. Then, click on the `User guides, package vignettes and other documentation.` link. Finally, click the `Rcpp::Rcpp-attributes` link.

### Describe the problem in detail

The user is prompted to download the associated PDF, rather than having it open in a PDF viewer.

### Describe the behavior you expected

We should open vignettes in the PDF viewer.

See also: https://community.rstudio.com/t/viewing-pdf-vignettes-in-preview-release/13877

This is a regression from v1.1, so marking at v1.2 milestone.

### PR DESCRIPTION
**Pull request by kevinushey**
This PR ensures that attempts to download `.pdf` files will download those files to a temporary directory and immediately open them, rather than prompting the user to download those files to a particular directory. This ensures that attempts to open vignettes from the help page work as expected, and brings us back in line with v1.1 behavior.

### CODE REVIEW

File: src/cpp/desktop/DesktopWebPage.cpp [hunk 1]
```diff
@@ -60,8 +61,57 @@ void onLinkHovered(const QString& url)
...
+void onPdfDownloadRequested(QWebEngineDownloadItem* downloadItem)
+{
+   QString scratchDir =
+         QString::fromStdString(options().scratchTempDir().absolutePath());
+   
+   // if we're requesting the download of a file with a '.pdf' extension,
+   // re-use that file name (since most desktop applications will display the
+   /// associated filename somewhere visible)
+   QString fileName = downloadItem->url().fileName();
+   if (fileName.endsWith(QStringLiteral(".pdf")))
```
Comment (jmcphers) [inline src/cpp/desktop/DesktopWebPage.cpp]: Is/should this be case-insensitive?

Comment (jmcphers) [reply src/cpp/desktop/DesktopWebPage.cpp]: We should also trim off the querystring here.

File: src/cpp/desktop/DesktopWebPage.cpp [hunk 2]
```diff
@@ -60,8 +61,57 @@ void onLinkHovered(const QString& url)
...
+                       const QString& downloadPath)
+{
+   QObject::connect(
+            downloadItem, &QWebEngineDownloadItem::finished,
+            [=]()
+   {
+      desktop::openFile(downloadPath);
+   });
+   
+   downloadItem->setPath(downloadPath);
```
Comment (jmcphers) [inline src/cpp/desktop/DesktopWebPage.cpp]: Does this accept `/` separators on Windows?

File: src/cpp/desktop/DesktopWebPage.cpp [hunk 3]
```diff
@@ -60,8 +61,57 @@ void onLinkHovered(const QString& url)
...
+   downloadItem->accept();
+}
+
+void onPdfDownloadRequested(QWebEngineDownloadItem* downloadItem)
+{
+   QString scratchDir =
+         QString::fromStdString(options().scratchTempDir().absolutePath());
```
Comment (MariaSemple) [inline src/cpp/desktop/DesktopWebPage.cpp]: It's not possible to pick a download location at download request time?

**Comment by kevinushey** [comment]
Indeed, `/` separators are accepted on Windows.

**Pull request merged**